test/test_eds.py

```diff
@@ -47,6 +48,20 @@ def test_record(self):
         self.assertEqual(var.data_type, canopen.objectdictionary.UNSIGNED32)
         self.assertEqual(var.access_type, 'ro')
 
+    def test_record_with_limits(self):
+        int8 = self.od[0x3020]
+        self.assertEqual(int8.min, 0)
+        self.assertEqual(int8.max, 127)
+        uint8 = self.od[0x3021]
+        self.assertEqual(uint8.min, 2)
+        self.assertEqual(uint8.max, 10)
+        int32 = self.od[0x3030]
+        self.assertEqual(int32.min, -2147483648)
```

Comment:
int32.min should be -1

Reply:
You are correct. I don't know why I didn't see it. This should be the correct function:

`def _signed_int_from_hex(hex_str, bit_length):
    number = int(hex_str, 0)
    max_value = (1 << (bit_length - 1)) - 1
    
    if number > max_value:
        return number - (1 << bit_length)
    else:
        return number`

I will commit a full fix with correct tests in the evening.